src/ui/server/helpers/task-list/generate-groups-and-tasks/prepare-application.ts

```diff
@@ -22,7 +43,12 @@ const createPrepareApplicationTasks = (referenceNumber: number, otherGroups: Tas
     href: `${INSURANCE_ROOT}/${referenceNumber}${POLICY_AND_EXPORTS.TYPE_OF_POLICY}`,
     title: TASKS.LIST.PREPARE_APPLICATION.TASKS.POLICY_TYPE_AND_EXPORTS,
     id: TASK_IDS.PREPARE_APPLICATION.POLICY_TYPE_AND_EXPORTS,
-    fields: Object.values(FIELD_IDS.INSURANCE.POLICY_AND_EXPORTS.TYPE_OF_POLICY),
+    fields: Object.values({
+      ...FIELD_IDS.INSURANCE.POLICY_AND_EXPORTS.TYPE_OF_POLICY,
+      ...SHARED_CONTRACT_POLICY,
+      ...getContractPolicyTasks(policyType),
+      ...FIELD_IDS.INSURANCE.POLICY_AND_EXPORTS.ABOUT_GOODS_OR_SERVICES,
+    }),
```

Comment:
@Zainzzkk This is all that's needed to add extra required fields to a task list group 😊 🎉

So the overall logic (in another function), relies on `fields` to be the definitive list of fields that are required in order for a task to be marked as "completed". It checks if all the fields are present in the saved data. Similar for "in progress" or "not started yet" 👍